core-web/libs/portlets/edit-ema/portlet/src/lib/components/dot-ema-dialog/dot-ema-dialog.component.ts

```diff
@@ -0,0 +1,192 @@
+import { fromEvent } from 'rxjs';
+
+import { NgIf, NgStyle, NgSwitch, NgSwitchCase } from '@angular/common';
+import {
+    ChangeDetectionStrategy,
+    Component,
+    DestroyRef,
+    ElementRef,
+    EventEmitter,
+    Output,
+    ViewChild,
+    inject
+} from '@angular/core';
+import { takeUntilDestroyed, toSignal } from '@angular/core/rxjs-interop';
+
+import { DialogModule } from 'primeng/dialog';
+
+import { DotCMSBaseTypesContentTypes } from '@dotcms/dotcms-models';
+import { DotSpinnerModule, SafeUrlPipe } from '@dotcms/ui';
+
+import { DotEmaDialogStore } from './store/dot-ema-dialog.store';
+
+import { ActionPayload } from '../../shared/models';
+import { EmaFormSelectorComponent } from '../ema-form-selector/ema-form-selector.component';
+
+@Component({
+    selector: 'dot-edit-ema-dialog',
+    standalone: true,
+    templateUrl: './dot-ema-dialog.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    imports: [
+        NgIf,
+        NgSwitch,
+        NgSwitchCase,
+        NgStyle,
+        SafeUrlPipe,
+        EmaFormSelectorComponent,
+        DialogModule,
+        DotSpinnerModule
+    ],
+    providers: [DotEmaDialogStore]
+})
+export class DotEmaDialogComponent {
+    @ViewChild('iframe') iframe: ElementRef<HTMLIFrameElement>;
+
+    @Output() customEvent = new EventEmitter<{ event: CustomEvent; payload: ActionPayload }>();
```

Comment:
```suggestion
    @Output() action = new EventEmitter<{ event: CustomEvent; payload: ActionPayload }>();
```

The fact that this is comming from a `CustomEvent` in the JSP should be a driver for the name of the event in the component API.

Reply:
@zJaaal check this one out.